cryptomator/ios
# Remove dependency to CryptomatorFileProvider from main target

Removes the dependency on the `CryptomatorFileProvider` framework from the main target by enabling / disabling the maintenance mode via XPC.
Furthermore, the `purposeIdentifier` will be used when doing coordinated reads and writes to the FileProvider document storage to prevent triggering `startProvidingItem(at:completionHandler:)` / `itemChanged(at:)` when reading or writing an item inside a FileProvider operation or the creation / opening of the database connection.

How to test:
- check that the main target does no longer depend on the `CryptomatorFileProvider` framework
- check that the operations rename vault, move vault and change vault password still require no running operations and lock the corresponding vault
- check that the `deleteItem(at:)` FileProvider operation does no longer trigger `itemChanged(at:)`
